lib/rubygems/specification_policy.rb

```diff
@@ -33,13 +33,32 @@ def initialize(specification)
   attr_accessor :packaging
 
   ##
-  # Checks that the specification contains all required fields, and does a
-  # very basic sanity check.
+  # Does a sanity check on the specification.
   #
   # Raises InvalidSpecificationException if the spec does not pass the
   # checks.
+  #
+  # It also performs some validations that do not raise but print warning
+  # messages instead.
 
   def validate(strict = false)
+    validate_strict
```

Comment:
we will `validate_strict` even if `strict` is false?

Reply:
Nice catch, the method was very poorly named!

I'll rename this method to `validate_required!` to avoid any confusion with the `strict` flag.

Reply:
The answer to your question is "yes", by the way :)

Reply:
To expand on the answer, the `strict` flag means that validations that normally lead to warnings should be considered as hard errors, but validations that always lead to hard errors should still be run even if `strict` is `false`.

Reply:
Code updated!